client.go

```diff
@@ -113,9 +117,43 @@ func (c *PingClient) PingMutualTLS(addr string) (*pb.Pong, error) {
 	return c.Ping(addr)
 }
 
+func MutualTLS(addr string) (*grpc.ClientConn, error) {
+
+	// Load the certificates from disk
+	certificate, err := tls.LoadX509KeyPair(ClientCert, ClientKey)
+	if err != nil {
+		return nil, fmt.Errorf("could not load client key pair: %s", err)
+	}
+
+	// Create a certificate pool from the certificate authority
+	certPool := x509.NewCertPool()
+	ca, err := ioutil.ReadFile(ExampleCA)
+	if err != nil {
+		return nil, fmt.Errorf("could not read ca certificate: %s", err)
+	}
+
+	// Append the client certificates from the CA
+	if ok := certPool.AppendCertsFromPEM(ca); !ok {
+		return nil, errors.New("failed to append ca certs")
+	}
+
+	// Create the TLS credentials for transport
+	creds := credentials.NewTLS(&tls.Config{
+		ServerName:   ServerName,
+		Certificates: []tls.Certificate{certificate},
+		RootCAs:      certPool,
+	})
+
+	conn, err := grpc.Dial(addr, grpc.WithTransportCredentials(creds))
+	if err != nil {
+		return nil, fmt.Errorf("could not connect to %s: %s", addr, err)
+	}
+	return conn, nil
+}
+
 // PingTLS is a helper method for server-side encryption that does not expect
 // client authentication or credentials. It is mostly here for benchmarking.
-func (c *PingClient) PingTLS(addr string) (*pb.Pong, error) {
+func TLS(addr string) (*grpc.ClientConn, error) {
```

Comment:
I think I'd prefer this to be renamed ServerTLS to make it more even with MutualTLS

Reply:
There is also die Option to add `Dialer` to those `func`s to give them names like `DialTLS` or `DialInsecure` or so?

Reply:
Yeh, I like that idea!